lib/server/swarm.js

```diff
@@ -7,12 +7,27 @@ var randomIterate = require('random-iterate')
 // Regard this as the default implementation of an interface that you
 // need to support when overriding Server.createSwarm() and Server.getSwarm()
 function Swarm (infoHash, server) {
+  this.complete = 0
+  this.incomplete = 0
   this.peers = new LRU({
     max: server.peersCacheLength || 1000,
-    maxAge: server.peersCacheTtl || 900000 // 900 000ms = 15 minutes
+    maxAge: server.peersCacheTtl || 20 * 60 * 1000 // 20 minutes
+  })
+
+  // When a websocket peer is evicted from the LRU cache, close the websocket
+  // after a short timeout period. We wait 1s so the server has a chance to send
+  // a response to 'stopped' events, which remove the peer and cause an eviction.
+  this.peers.on('evict', function (data) {
+    var peer = data.value
+    if (peer.socket) {
+      setTimeout(function () {
+        try {
+          peer.socket.close()
```

Comment:
LRU .remove method does not emit evict. See [chriso/lru/index.js#L30](https://github.com/chriso/lru/blob/master/index.js#L30)

But if the peer is getting evicted because of LRU limit, it will be evicted and stopped message won't ever be received, because we are not receiving anything. We should perform the stuff done in the event of stoppped message when the eviction happens. No need for timeout. I would however close and nullify on nextTick.